py3status/modules/hueshift.py

```diff
@@ -0,0 +1,185 @@
+    coffee free all nighter: 8000
+
+Examples:
+```
+# good rules; show command name; don't colorize everything
+hueshift {
+    format = '[\?if=pgrep&color=bad \u263c!|[\?color=hue&show \u263c '
+    format += '{command}] {temperature}K]'
+    minimum = 3000
+    maximum = 10000
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+# warm
+{'full_text': u'\u263c 3000K', 'color': #f3c},
+
+# neutral
+{'full_text': u'\u263c 6500K', 'color': #ff3}
+
+# cool
+{'full_text': u'\u263c 10000K', 'color': #3cf},
+
+paused
+{'full_text': u'\u263c!', 'color': #f00}
+"""
+
+STRING_NOT_INSTALLED = 'not installed'
+STRING_BAD_COMMAND = 'invalid command'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_down = 5
+    button_reset = 3
+    button_up = 4
+    command = None
+    delta = 250
+    format = '\?if=pgrep&color=bad \u263c!|\?color=hue \u263c {temperature}K'
+    initial = 6500
+    maximum = 25000
+    minimum = 1000
+    thresholds = [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]
+
+    def post_config_hook(self):
+        hueshift_commands = ['blueshift', 'redshift', 'sct']
+        self.pgrep_command = ['pgrep', '|'.join(hueshift_commands)]
+        if not self.command:
+            self.command = self.py3.check_commands(hueshift_commands)
+        elif self.command not in hueshift_commands:
+            raise Exception(STRING_BAD_COMMAND)
+        elif not self.py3.check_commands(self.command):
+            raise Exception('%s %s' % (self.command, STRING_NOT_INSTALLED))
+        if not self.command:
+            raise Exception(STRING_NOT_INSTALLED)
+
+        self.hue = {
+            'blueshift': {
+                'set': lambda value: ['blueshift', '-p', '-t', str(value)],
+                'reset': lambda value: ['blueshift', '-p', '-r'],
+            },
+            'redshift': {
+                'set': lambda value: ['redshift', '-r', '-O', str(value)],
+                'reset': lambda value: ['redshift', '-r', '-x'],
+            },
+            'sct': {
+                'set': lambda value: ['sct', str(value)],
+                'reset': lambda value: ['sct']
+            },
+        }
+
+        self.last_command = None
+        self.temperature = 6500
+        self._pgrep()
+
+        if self.minimum < 1000:
+            self.minimum = 1000
+        if self.maximum > 25000:
+            self.maximum = 25000
+        if self.command == 'sct' and self.maximum > 10000:
+            self.maximum = 10000
+        if not(self.minimum <= self.initial <= self.maximum):
+            self.initial = 6500
+        if not self.pgrep:
+            self._command('set', self.initial)
+
+    def _command(self, action=None, value=None):
+        command = self.hue[self.command][action](value)
+        if command != self.last_command:
+            self.py3.command_run(command)
+            self.last_command = command
+        else:
+            self.py3.prevent_refresh()
+
+    def _pgrep(self):
```

Comment:
it is nice to have functions that you use defined before they are used as it makes reading the code easier :)

Reply:
the function name should give me some idea what it does too.  Here it sets `self.pgrep` so maybe call it `_set_pgrep()` also some docstring or comment might be nice to about what you are doing.

but really `pgrep` is a silly name `is_rival_color_setter_running()` and returning a bool would be much nicer

Reply:
Changed to `update_pgrep_boolean`. I'll look at this storage stuffs next day as I need some sleep first. I know I use `initial` param for `psc` and it's the same param used to reset the shift whenever users want to.

Toggling sounds nicer than resetting.

Reply:
pgrep is a horrible name for a variable, please change it to anything

Reply:
Done. No more `pgrep`. Storage temperature is nice. ;-)